### ISSUE TITLE
[FR] Support Keras core (multi-backend Keras)

### ISSUE THREAD

**Issue by chenmoneygithub** [mlflow/mlflow#9313]
Keras recently went multi-backend, users can write PyTorch, JAX and Tensorflow with the same code in Keras. The new repo is called [Keras Core](https://github.com/keras-team/keras-core). 

We might want to consider supporting it, and one main motivation is the JAX support. JAX has been out for a while, but it's not popular outside Google because lack of documentation. Keras ideally could fill in that gap, so it would be nice to do integration within MLflow. JAX comes with some nice features such as seamless Numpy integration, good support for model parallelism, and most significantly, its performance is better than TF and Torch.

To support Keras Core in Mlflow, there are 2 main parts:
- Logging/Tracking
    - `mlflow.autolog()` should continue to be supported for consistency, which proactively pull the metrics, like training loss..
    -  For better control, we can implement mlflow callbacks so that users can plug into their training, e.g., `MlflowCallback` which 
        logs data into mlflow server.
- Model registry/serving: could be the same as the underlying framework, Tensorflow/JAX/PyTorch.

Not something of our top-1 priority, but worth checking out if we have bandwidth.

**Comment by chenmoneygithub** [comment]
As a reference, weights & bias has made prototype towards Keras core: https://github.com/wandb/wandb/issues/5939

**Comment by WeichenXu123** [comment]
This is worth checking, 

But looks like keras-core is still experimental, we can keep an eye on it but let's wait it releasing stable version first.
